libobs/obs-output.c

```diff
@@ -1678,16 +1679,29 @@ bool obs_output_can_begin_data_capture(const obs_output_t *output,
 		pthread_join(output->end_data_capture_thread, NULL);
 
 	convert_flags(output, flags, &encoded, &has_video, &has_audio,
-			&has_service);
+			&has_service, &force_encoder);
 
 	return can_begin_data_capture(output, encoded, has_video, has_audio,
 			has_service);
 }
 
+static inline void ensure_force_initialize_encoder(obs_encoder_t* encoder)
+{
+	if (encoder) {
```

Comment:
_encoder_ is checked to be not null in all places where that function called. 
So no need to check it again inside function. 

Can use `assert( encoder != nullptr );` in case someone decide to call function without checking encoder.

Reply:
Agree, check removed!